httpclient5/src/main/java/org/apache/hc/client5/http/impl/classic/InternalHttpClient.java

```diff
@@ -161,11 +161,15 @@ protected CloseableHttpResponse doExecute(
             setupContext(localcontext);
 
             final HttpHost resolvedTarget = target != null ? target : RoutingSupport.determineHost(request);
-            if (request.getScheme() == null) {
-                request.setScheme(resolvedTarget.getSchemeName());
-            }
-            if (request.getAuthority() == null) {
-                request.setAuthority(new URIAuthority(resolvedTarget));
+            if (resolvedTarget != null) {
```

Comment:
HI @arturobernalg 👋 
We could throw a more precise exception using `Objects.requireNonNull(...)`, which would reduce the nesting.